git-cliff-core/src/repo.rs

```diff
@@ -75,37 +85,152 @@ impl Repository {
 			.filter_map(|id| self.inner.find_commit(id).ok())
 			.collect();
 		if include_path.is_some() || exclude_path.is_some() {
+			// Normalize the glob patterns
+			let include_patterns = include_path.map(|patterns| {
+				patterns.into_iter().map(Self::normalize_glob).collect()
+			});
+			let exclude_patterns = exclude_path.map(|patterns| {
+				patterns.into_iter().map(Self::normalize_glob).collect()
+			});
+
 			commits.retain(|commit| {
-				if let Ok(prev_commit) = commit.parent(0) {
-					if let Ok(diff) = self.inner.diff_tree_to_tree(
-						commit.tree().ok().as_ref(),
-						prev_commit.tree().ok().as_ref(),
-						None,
-					) {
-						return diff
-							.deltas()
-							.filter_map(|delta| delta.new_file().path())
-							.any(|new_file_path| {
-								if let Some(include_path) = &include_path {
-									include_path
-										.iter()
-										.any(|glob| glob.matches_path(new_file_path))
-								} else if let Some(exclude_path) = &exclude_path {
-									!exclude_path
-										.iter()
-										.any(|glob| glob.matches_path(new_file_path))
-								} else {
-									false
-								}
-							});
-					}
-				}
-				false
+				self.should_retain_commit(
+					commit,
+					&include_patterns,
+					&exclude_patterns,
+				)
 			});
 		}
 		Ok(commits)
 	}
 
+	/// Normalizes the glob pattern by removing the leading `./`.
+	fn normalize_glob(pattern: Pattern) -> Pattern {
+		match pattern.as_str().strip_prefix("./") {
+			Some(stripped) => Pattern::new(stripped)
+				.expect("Removing the leading ./ will not fail"),
+			None => pattern,
+		}
+	}
+
+	/// Calculates whether the commit should be retained or not.
+	fn should_retain_commit(
```

Comment:
Can you add unit tests for this?

Reply:
Added a comprehensive test suit in [`33f56cf` (#772)](https://github.com/orhun/git-cliff/pull/772/commits/33f56cf2d916106168afc5b21ec8685c573304bf)